src/find/matchers/time.rs

```diff
@@ -250,44 +260,68 @@ impl Matcher for FileTimeMatcher {
 impl FileTimeMatcher {
     /// Implementation of matches that returns a result, allowing use to use try!
     /// to deal with the errors.
-    fn matches_impl(&self, file_info: &DirEntry, now: SystemTime) -> Result<bool, Box<dyn Error>> {
+    fn matches_impl(
+        &self,
+        file_info: &DirEntry,
+        start_time: SystemTime,
+    ) -> Result<bool, Box<dyn Error>> {
         let this_time = self.file_time_type.get_file_time(file_info.metadata()?)?;
         let mut is_negative = false;
         // durations can't be negative. So duration_since returns a duration
         // wrapped in an error if now < this_time.
-        let age = match now.duration_since(this_time) {
+        let age = match start_time.duration_since(this_time) {
             Ok(duration) => duration,
             Err(e) => {
                 is_negative = true;
                 e.duration()
             }
         };
         let age_in_seconds: i64 = age.as_secs() as i64 * if is_negative { -1 } else { 1 };
+
         // rust division truncates towards zero (see
         // https://github.com/rust-lang/rust/blob/master/src/libcore/ops.rs#L580 )
         // so a simple age_in_seconds / SECONDS_PER_DAY gives the wrong answer
         // for negative ages: a file whose age is 1 second in the future needs to
         // count as -1 day old, not 0.
-        let age_in_days = age_in_seconds / SECONDS_PER_DAY + if is_negative { -1 } else { 0 };
+        // If today_start is true, we should count it as 0 days old.
+        // because today is 00:00:00, so we need to subtract 1 day.
+        let negative_offset = if is_negative && !self.today_start {
+            -1
+        } else {
+            0
+        };
+
+        let age_in_days = age_in_seconds / SECONDS_PER_DAY + negative_offset;
         Ok(self.days.imatches(age_in_days))
     }
 
-    pub fn new(file_time_type: FileTimeType, days: ComparableValue) -> Self {
+    pub fn new(file_time_type: FileTimeType, days: ComparableValue, today_start: bool) -> Self {
         Self {
             days,
             file_time_type,
+            today_start,
         }
     }
 }
 
 pub struct FileAgeRangeMatcher {
     minutes: ComparableValue,
     file_time_type: FileTimeType,
+    today_start: bool,
 }
 
 impl Matcher for FileAgeRangeMatcher {
     fn matches(&self, file_info: &DirEntry, matcher_io: &mut MatcherIO) -> bool {
-        match self.matches_impl(file_info, matcher_io.now()) {
+        let time = if self.today_start {
```

Comment:
maybe move this into a function as you are duplicating it?

Reply:
Thanks, it would be better to wrap it in a function. Changed in the latest commit.